table/block_based_table_reader.cc

```diff
@@ -430,7 +429,10 @@ struct BlockBasedTable::Rep {
   // the LRU cache will never push flush them out, hence they're pinned
   CachableEntry<FilterBlockReader> filter_entry;
   CachableEntry<IndexReader> index_entry;
-  unique_ptr<Block> range_del_block;
+  // range deletion meta-block is pinned through reader's lifetime when LRU
+  // cache is enabled.
+  CachableEntry<Block> range_del_entry;
+  BlockHandle range_del_handle;
```

Comment:
Aghh got it.  I thought the above lines are inside same function but they're not.  Now I believe phabricator gives us a better view :p